s3/src/main/scala/tamer/s3/S3Configuration.scala

```diff
@@ -0,0 +1,119 @@
+package tamer
+package s3
+
+import com.sksamuel.avro4s.{Codec, SchemaFor}
+import eu.timepit.refined.types.numeric.PosInt
+import zio.stream.ZTransducer
+import zio.{Queue, UIO, ZIO}
+
+import java.time.format.DateTimeFormatter
+import java.time.{Duration, Instant}
+import scala.util.hashing.MurmurHash3.stringHash
+
+
+final case class S3Configuration[
+  K <: Product : Codec : SchemaFor,
+  V <: Product : Codec : SchemaFor,
+  S <: Product : Codec : SchemaFor
+](
+   bucketName: String,
+   prefix: String,
+   tamerStateKafkaRecordKey: Int,
+   transducer: ZTransducer[Any, TamerError, Byte, V],
+   parallelism: PosInt,
+   timeouts: S3Configuration.Timeouts,
+   transitions: S3Configuration.StateTransitions[K, V, S],
+ ) {
+  val generic: SourceConfiguration[K, V, S] = SourceConfiguration[K, V, S](
+    SourceConfiguration.SourceSerde[K, V, S](),
+    defaultState = transitions.initialState,
+    tamerStateKafkaRecordKey = tamerStateKafkaRecordKey,
+    S3Configuration.this.toString,
+  )
+}
+
+object S3Configuration {
+
+  case class Timeouts(
+                       minimumIntervalForBucketFetch: Duration,
+                       maximumIntervalForBucketFetch: Duration,
+                     )
```

Comment:
First, let mes say that in the target state, these should be so important, because the user should be empowered to chose the desired schedule.
But I would not call them timeouts, since they are not interrupting anything. This is the interval of time between bucket fetches to retrieve an updated file list. It is expressed as a range.
Grouping makes sense but I would call this `KeysFetchInterval` or something to this effect.